pickhardtpayments/ChannelGraph.py

```diff
@@ -13,22 +13,29 @@ class ChannelGraph():
     contain parallel channels.
     """
 
-    def _get_channel_json(self, filename: str):
+    def _get_channel_json(self, filename: str, fmt: str = "cln"):
         """
         extracts the dictionary from the file that contains lightnig-cli listchannels json string
         """
-        f = open(filename)
-        return json.load(f)["channels"]
+        with open(filename) as f:
```

Comment:
just curiosity is there any rational why the `with ... as ...` syntax is preferable?

Reply:
Will try to get used doing this. this is an excellent reason!